src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -320,6 +323,61 @@ private static string GetExpectedRedirectUri(string clientId)
         }
 
         /// <summary>
+        /// Converts to MSAL Account Id or Null
+        /// </summary>
+        /// <param name="wamAccount"></param>
+        /// <param name="clientID"></param>
+        /// <param name="logger"></param>
+        public static IAccount ConvertToMsalAccount(
+                NativeInterop.Account wamAccount, 
+                string clientID,
+                ILoggerAdapter logger)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(wamAccount.AccountId) ||
+                    string.IsNullOrEmpty(wamAccount.HomeAccountid) ||
+                    string.IsNullOrEmpty(wamAccount.Environment) ||
+                    string.IsNullOrEmpty(wamAccount.UserName))
+                {
+                    // Create PII enabled string builder
+                    var builder = new StringBuilder(
```

Comment:
nit: for logging, I like to create extension method, as it can be reused.

void ToLogMessage(this WamAccount, logger)